### PR TITLE
feat(connectRefinementList): update getWidgetSearchParameters 

### PR DESCRIPTION
**Pull request by samouss**
This PR updates the lifecycle `getWidgetSearchParameters` in `connectRefinementList` to add the configuration of facets and default value. We don't apply the previous value on purpose, the `uiState` drives completely the `SearchParameters`.

I've not removed the `getConfiguration` lifecycle (yet) to avoid breaking everything. Once we've made the switch to `getWidgetSearchParameters` we'll remove them.

### CODE REVIEW

File: src/connectors/refinement-list/__tests__/connectRefinementList-test.js [hunk 1]
```diff
@@ -2080,4 +1969,370 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/refinement-
...
+
+    test('should add an entry equal to the refinement', () => {
+      const [widget, helper, refine] = getInitializedWidget();
+      refine('value');
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
+      });
+      expect(uiStateAfter).toMatchSnapshot();
```
Comment (francoischalifour) [inline src/connectors/refinement-list/__tests__/connectRefinementList-test.js]: Same comment as other PRs about snapshots.

File: src/connectors/refinement-list/__tests__/connectRefinementList-test.js [hunk 2]
```diff
@@ -2080,4 +1969,370 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/refinement-
...
+      const uiStateBefore = {
+        refinementList: {
+          otherFacetAttribute: ['val'],
+        },
+      };
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
+      });
+      expect(uiStateAfter).toMatchSnapshot();
```
Comment (francoischalifour) [inline src/connectors/refinement-list/__tests__/connectRefinementList-test.js]: Same comment as other PRs about snapshots.

File: src/connectors/refinement-list/__tests__/connectRefinementList-test.js (hunk 1 above)
Comment (samouss) [reply src/connectors/refinement-list/__tests__/connectRefinementList-test.js]: https://github.com/algolia/instantsearch.js/pull/4010/commits/c25b9c4a66c1821c56cf0a8ed2c10454112483bb

File: src/connectors/refinement-list/__tests__/connectRefinementList-test.js (hunk 2 above)
Comment (samouss) [reply src/connectors/refinement-list/__tests__/connectRefinementList-test.js]: https://github.com/algolia/instantsearch.js/pull/4010/commits/c25b9c4a66c1821c56cf0a8ed2c10454112483bb

**Pull request merged**